src/Operators/finitedifference.jl

```diff
@@ -1760,6 +1768,86 @@ function stencil_right_boundary(
     bc.val
 end
 
+
+
+abstract type CurlFiniteDifferenceOperator <: FiniteDifferenceOperator end
+return_eltype(::CurlFiniteDifferenceOperator, arg) =
+    Geometry.curl_result_type(Val((3,)), eltype(arg))
+
+"""
+    C = CurlC2F(;boundaryname=boundarycondition...)
+    C.(v)
+
+Compute the vertical contribution to the curl of a center-valued field vector
+`v`, returning a face-valued scalar field, using the stencil
+```math
+\\begin{align*}
+C(v)[i]^1 = - \\frac{1}{J[i]} (v₂[i+\\tfrac{1}{2}] - v₂[i-\\tfrac{1}{2}]) \\\\
+C(v)[i]^2 = \\frac{1}{J[i]}  (v₁[i+\\tfrac{1}{2}] - v₁[i-\\tfrac{1}{2}])
+\\end{align*}
+```
+where ``v₁` and ``v₂`` are the 1st and 2nd covariant components of ``v``, and
+``J`` is the Jacobian determinant.
+"""
+struct CurlC2F{BC} <: CurlFiniteDifferenceOperator
```

Comment:
I'm a bit confused about the docs here. If this is the vertical contribution of the curl, why would we need BCs? Shouldn't it be using points at the same radius (on the sphere), and it should be a collocated operation, working across DG points.

If it's for the horizontal components, then I think this should be okay.

Reply:
Actually, if it's for horizontal components, shouldn't this live in the DG section?

Reply:
Okay, perhaps the docs aren't clear here: it's the contribution of the derivative in the vertical direction, i.e. the terms involve d/d\xi_3. But it only applies to horizontal vectors (i.e. `Covariant12Vector`s)

Reply:
Curls are confusing.

Reply:
"compute the vertical contribution to the horizontal curl"?

Reply:
Hmm. What is the direction of the output curl vector?

Reply:
> "compute the vertical contribution to the horizontal curl"?

That sounds much more clear to me.